scripts/filter_multimappers.py

```diff
@@ -1,43 +1,51 @@
-respectively.
-Finally, if there are any remaining alignments in the current alignments
-list and selects the best alignment(s) for those queries, writing them to
-standard output. If --nh is set, all query names are modified to include the
-NH tag after an underscore.
-
-Input:
-    input_sam_file: Path to the input SAM file, ordered by query name.
-
-Output:
-    standard output: Filtered SAM file
-
-Functions:
-    parse_arguments():
-        Command-line arguments parser
-    count_indels(pysam.AlignedSegment):
-        Count the number of indels in an alignment based on its CIGAR string.
-    find_best_alignments(list[pysam.AlignedSegment]):
-        Find alignments with less indels
-    write_output(list[pysam.AlignedSegment]):
-        Write the output to the standard output (stdout).
-    main(Path):
-        Filter multimappers by indels count.
-
-Usage:
-    filter_multimappers.py SAM
-    filter_multimappers.py SAM > SAM
-"""
+# pylint: disable=line-too-long
+
+"""Filter miRNA reads mapped to multiple locations by indel count.
+
+From all the alignments with the same query name and edit distance, keep the
+ones with the higher amount of indels. Supplementary alignments are dismissed.
+Additionally, the 'NH' and 'HI' tags are updated to match the new amount of
+alignments and keep their identifier within the new set respectively. If the
+CLI flag `--nh` is set, query names are appended the suffix '_#' were '#' is
+the updated alignment's NH tag.
+
+The following assumptions are made:
+    - The input SAM file is sorted by query name.
+
+
+EXAMPLES:
+    Example 1
+    IN SAM records:
+        read-1	16	19	77595	255	14M1D8M	*	0	0	GCAGGAGAATCACTGATGTCAG	*	MD:Z:14^T2A1C3	NH:i:2	NM:i:3	XA:Z:Q	XI:i:1
+        read-1	0	19	330456	255	4M1D1M1I3M1D13M	*	0	0	CTGACATCAGTGATTCTCCTGC	*	MD:Z:4^G4^A13	NH:i:2	NM:i:3	XA:Z:Q	XI:i:0
+    command:
+        filter_multimappers.py SAM > out_SAM
+    OUT SAM record:
+        read-1	0	19	330456	255	4M1D1M1I3M1D13M	*	0	0	CTGACATCAGTGATTCTCCTGC	*	MD:Z:4^G4^A13	NH:i:1	HI:i:1	NM:i:3	XA:Z:Q	XI:i:0
+
+    Example 2
+    IN SAM records:
+        read-2	0	19	142777	255	21M	*	0	0	GCTAGGTGGGAGGCTTGAAGCT	*	MD:Z:4C0T14A	NH:i:3	NM:i:3	XA:Z:Q	XI:i:0
+        read-2	16	19	270081	255	6M1I14M	*	0	0	GCTTCAAGCCTCCCACCTAGC	*	MD:Z:14G0G4	NH:i:3	NM:i:3	XA:Z:Q	XI:i:2
+        read-2	16	19	545543	255	6M1I14M	*	0	0	GCTTCAAGCCTCCCACCTAGC	*	MD:Z:14A0G4	NH:i:3	NM:i:3	XA:Z:Q	XI:i:1
+    command:
+        filter_multimappers.py SAM --nh > out_SAM
+    OUT SAM record:
+        read-2_2	16	19	270081	255	6M1I14M	*	0	0	GCTTCAAGCCTCCCACCTAGC	*	MD:Z:14G0G4	NH:i:2	HI:i:1	NM:i:3	XA:Z:Q	XI:i:2
+        read-2_2	16	19	545543	255	6M1I14M	*	0	0	GCTTCAAGCCTCCCACCTAGC	*	MD:Z:14A0G4	NH:i:2	HI:i:2	NM:i:3	XA:Z:Q	XI:i:1
+
+    Example 3
+    IN SAM records:
+        read-3	2128	19	63251	255	22M	*	0	0	AAAGCGCTTCCCTTCAGAGTGA	*	MD:Z:21T	NH:i:1	NM:i:1
+    command:
+        filter_multimappers.py SAM > out_SAM
+    OUT SAM record:
```

Comment:
I think it would help tremendously if you would put in the actual alignments for each read and the reference, e.g., for example 1:

```
GCAGGAGAATCACTTGAAGCCAG
|||||||||||||| || | ||| (1 InDel, 2 mismatches; discarded)
GCAGGAGAATCACT-GATGTCAG

CTGAGC-TCAAGTGATTCTCCTGC
|||| | ||| ||||||||||||| (3 InDels, 0 mismatches; retained)
CTGA-CATCA-GTGATTCTCCTGC
```